ledger-tool/src/main.rs

```diff
@@ -888,11 +888,31 @@ fn load_bank_forks(
     }
 
     let account_paths = if let Some(account_paths) = arg_matches.value_of("account_paths") {
+        // If this blockstore access is Primary, no other process (solana-validator) can hold
+        // Primary access. So, allow a custom accounts path without worry of wiping the accounts
+        // of solana-validator.
         if !blockstore.is_primary_access() {
-            // Be defensive, when default account dir is explicitly specified, it's still possible
-            // to wipe the dir possibly shared by the running validator!
-            eprintln!("Error: custom accounts path is not supported under secondary access");
-            exit(1);
+            // Attempt to open the Blockstore in Primary access; if successful, no other process
+            // was holding Primary so allow things to proceed with custom accounts path. Release
+            // the Primary access instead of holding it to give priority to solana-validator over
```

Comment:
Not holding `Primary` was somewhat by design to give the `solana-validator` the ability to "override" `solana-ledger-tool`. But, you're right, this check as implemented isn't very thorough and could be race condition that would allow `solana-validator` to clobber `solana-ledger-tool` if they use the same, custom `--accounts` path.

This is mostly an internal efficiency thing, altho with v1.14 having accounts-index in memory by default, the speedup from this change is more marginal.